### PR TITLE
Add wasdat code controller

### PR DESCRIPTION
**Pull request by Maartenwut**
## Description

Add the wasdat code controller, it's very similar to the wasdat.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/maartenwut/wasdat_code/config.h [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4705
+#define PRODUCT_ID      0xC574
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Maartenwut
+#define PRODUCT         Wasdat
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/config.h:27]:
```suggestion
#define PRODUCT         Wasdat CODE
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+	setPinOutput(D3);
+	setPinOutput(F7);
+	setPinOutput(F5);
+	setPinOutput(F6);
+	setPinOutput(F1);
+	setPinOutput(F4);
+	setPinOutput(F0);
+	setPinOutput(D3);
+	setPinOutput(D5);
+	setPinOutput(D4);
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:326]:
```suggestion
    setPinOutput(D0);
    setPinOutput(D1);
    setPinOutput(D2);
    setPinOutput(D3);
    setPinOutput(F7);
    setPinOutput(F5);
    setPinOutput(F6);
    setPinOutput(F1);
    setPinOutput(F4);
    setPinOutput(F0);
    setPinOutput(D3);
    setPinOutput(D5);
    setPinOutput(D4);
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+	writePinHigh(F4);
+	writePinHigh(F0);
+	writePinHigh(D3);
+	writePinHigh(D5);
+
+    //Demultiplexer
+	writePinLow(D4);
+	writePinLow(D2);
+	writePinLow(D1);
+	writePinLow(D0);
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:306]:
```suggestion
    writePinLow(D4);
    writePinLow(D2);
    writePinLow(D1);
    writePinLow(D0);
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+static void unselect_cols(void) {
+    //Native
+	writePinHigh(F7);
+	writePinHigh(F5);
+	writePinHigh(F6);
+	writePinHigh(F1);
+	writePinHigh(F4);
+	writePinHigh(F0);
+	writePinHigh(D3);
+	writePinHigh(D5);
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:300]:
```suggestion
    writePinHigh(F7);
    writePinHigh(F5);
    writePinHigh(F6);
    writePinHigh(F1);
    writePinHigh(F4);
    writePinHigh(F0);
    writePinHigh(D3);
    writePinHigh(D5);
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+            break;
+        case 14:
+		//MUX
+            writePinLow(D4);
+            writePinLow(D2);
+            writePinLow(D1);
+            writePinLow(D0);
+            break;
+        case 15:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:282]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+            break;
+        case 13:
+		//MUX
+            writePinLow(D4);
+            writePinLow(D2);
+            writePinLow(D1);
+            writePinLow(D0);
+            break;
+        case 14:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:275]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+            break;
+        case 12:
+		//MUX
+            writePinLow(D4);
+            writePinLow(D2);
+            writePinLow(D1);
+            writePinLow(D0);
+            break;
+        case 13:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:268]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+            break;
+        case 11:
+		//MUX
+            writePinLow(D4);
+            writePinLow(D2);
+            writePinLow(D1);
+            writePinLow(D0);
+            break;
+        case 12:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:261]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 9]
```diff
@@ -0,0 +1,385 @@
...
+            break;
+        case 10:
+		//MUX
+            writePinLow(D4);
+            writePinLow(D2);
+            writePinLow(D1);
+            writePinLow(D0);
+            break;
+        case 11:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:254]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 10]
```diff
@@ -0,0 +1,385 @@
...
+            break;
+        case 9:
+		//MUX
+            writePinLow(D4);
+            writePinLow(D2);
+            writePinLow(D1);
+            writePinLow(D0);
+            break;
+        case 10:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:247]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 11]
```diff
@@ -0,0 +1,385 @@
...
+            writePinLow(D0);
+            break;
+        case 7:
+            writePinHigh(D5);
+            break;
+        case 8:
+            writePinHigh(D3);
+            break;
+        case 9:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:240]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 12]
```diff
@@ -0,0 +1,385 @@
...
+            writePinHigh(F1);
+            break;
+        case 4:
+            writePinHigh(F4);
+            break;
+        case 5:
+            writePinHigh(F0);
+            break;
+        case 6:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:227]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 13]
```diff
@@ -0,0 +1,385 @@
...
+            writePinHigh(D4);
+            writePinHigh(D1);
+            break;
+        case 14:
+		//MUX
+            writePinHigh(D4);
+            writePinHigh(D2);
+            break;
+        case 15:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:200]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 14]
```diff
@@ -0,0 +1,385 @@
...
+            writePinHigh(D2);
+            writePinHigh(D1);
+            break;
+        case 13:
+		//MUX
+            writePinHigh(D4);
+            writePinHigh(D1);
+            break;
+        case 14:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:195]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 15]
```diff
@@ -0,0 +1,385 @@
...
+            writePinHigh(D0);
+            break;
+        case 12:
+		//MUX
+            writePinHigh(D4);
+            writePinHigh(D2);
+            writePinHigh(D1);
+            break;
+        case 13:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:190]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 16]
```diff
@@ -0,0 +1,385 @@
...
+            writePinHigh(D2);
+            writePinHigh(D0);
+            break;
+        case 11:
+		//MUX
+            writePinHigh(D4);
+            writePinHigh(D0);
+            break;
+        case 12:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:184]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 17]
```diff
@@ -0,0 +1,385 @@
...
+            writePinHigh(D0);
+            break;
+        case 10:
+		//MUX
+            writePinHigh(D4);
+            writePinHigh(D2);
+            writePinHigh(D0);
+            break;
+        case 11:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:179]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 18]
```diff
@@ -0,0 +1,385 @@
...
+            writePinLow(D3);
+            break;
+        case 9:
+		//MUX
+            writePinHigh(D4);
+            writePinHigh(D1);
+            writePinHigh(D0);
+            break;
+        case 10:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:173]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 19]
```diff
@@ -0,0 +1,385 @@
...
+            writePinHigh(D0);
+            break;
+        case 7:
+            writePinLow(D5);
+            break;
+        case 8:
+            writePinLow(D3);
+            break;
+        case 9:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:167]:
```suggestion
```

File: keyboards/maartenwut/wasdat_code/matrix.c [hunk 20]
```diff
@@ -0,0 +1,385 @@
...
+            writePinLow(F1);
+            break;
+        case 4:
+            writePinLow(F4);
+            break;
+        case 5:
+            writePinLow(F0);
+            break;
+        case 6:
+		//MUX
```
Comment (fauxpark) [inline keyboards/maartenwut/wasdat_code/matrix.c:154]:
```suggestion
```

**Comment by Maartenwut** [comment]
> During the next breaking changes cycle we should refactor these two to `wasdat/v2` and `wasdat_code`.

Not sure what you mean here, could you elaborate?

**Comment by fauxpark** [comment]
Oops, fixed. Basically just a note to myself (or anyone else) that these two boards should probably go under a common `wasdat/` subfolder, as there is some shared config/code. But it'll be fine to merge this as it's currently named.

**Comment by Maartenwut** [comment]
Please merge?

File: keyboards/maartenwut/wasdat_code/info.json [hunk 21]
```diff
@@ -0,0 +1,459 @@
...
+                {"label":"D", "x":3.75, "y":3.5},
+                {"label":"F", "x":4.75, "y":3.5},
+                {"label":"G", "x":5.75, "y":3.5},
+                {"label":"H", "x":6.75, "y":3.5},
+                {"label":"J", "x":7.75, "y":3.5},
+                {"label":"K", "x":8.75, "y":3.5},
+                {"label":"L", "x":9.75, "y":3.5},
+                {"label":":", "x":10.75, "y":3.5},
+                {"label":"@", "x":11.75, "y":3.5},
+                {"label":"~", "x":12.75, "y":3.5},
```
Comment (noroadsleft) [inline keyboards/maartenwut/wasdat_code/info.json:211]: ISO Enter's object is out of sequence.

```suggestion

                {"label":"Delete", "x":15.25, "y":2.5},
                {"label":"End", "x":16.25, "y":2.5},
                {"label":"PgDn", "x":17.25, "y":2.5},

                {"label":"7", "x":18.5, "y":2.5},
                {"label":"8", "x":19.5, "y":2.5},
                {"label":"9", "x":20.5, "y":2.5},
                {"label":"+", "x":21.5, "y":2.5, "h":2},

                {"label":"Caps Lock", "x":0, "y":3.5, "w":1.75},
                {"label":"A", "x":1.75, "y":3.5},
                {"label":"S", "x":2.75, "y":3.5},
                {"label":"D", "x":3.75, "y":3.5},
                {"label":"F", "x":4.75, "y":3.5},
                {"label":"G", "x":5.75, "y":3.5},
                {"label":"H", "x":6.75, "y":3.5},
                {"label":"J", "x":7.75, "y":3.5},
                {"label":"K", "x":8.75, "y":3.5},
                {"label":"L", "x":9.75, "y":3.5},
                {"label":":", "x":10.75, "y":3.5},
                {"label":"@", "x":11.75, "y":3.5},
                {"label":"~", "x":12.75, "y":3.5},
                {"label":"Enter", "x":13.75, "y":2.5, "w":1.25, "h":2},
```

File: keyboards/maartenwut/wasdat_code/info.json [hunk 22]
```diff
@@ -0,0 +1,459 @@
...
```
Comment (noroadsleft) [inline keyboards/maartenwut/wasdat_code/info.json:426]: Same as above.

```suggestion

                {"label":"Delete", "x":15.25, "y":2.5},
                {"label":"End", "x":16.25, "y":2.5},
                {"label":"PgDn", "x":17.25, "y":2.5},

                {"label":"Caps Lock", "x":0, "y":3.5, "w":1.75},
                {"label":"A", "x":1.75, "y":3.5},
                {"label":"S", "x":2.75, "y":3.5},
                {"label":"D", "x":3.75, "y":3.5},
                {"label":"F", "x":4.75, "y":3.5},
                {"label":"G", "x":5.75, "y":3.5},
                {"label":"H", "x":6.75, "y":3.5},
                {"label":"J", "x":7.75, "y":3.5},
                {"label":"K", "x":8.75, "y":3.5},
                {"label":"L", "x":9.75, "y":3.5},
                {"label":":", "x":10.75, "y":3.5},
                {"label":"@", "x":11.75, "y":3.5},
                {"label":"~", "x":12.75, "y":3.5},
                {"label":"Enter", "x":13.75, "y":2.5, "w":1.25, "h":2},
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**